R/readData.R

```diff
@@ -1,19 +1,91 @@
 # ---------------------------------------------------------------------------- #
 # fast reading of big tables
-readTableFast<-function(filename,header=TRUE,skip=0,sep=""){
+readTableFast<-function(filename,header="auto",skip=0,sep="auto",as.datatable=FALSE, ...){
   
-  tab5rows <- read.table(filename, header = header,skip=skip,
-                         sep=sep, nrows = 100, stringsAsFactors=FALSE)
-  classes  <- sapply(tab5rows, class)
-  df = read.table(filename, 
-                  header = header,
-                  skip=skip,
-                  sep=sep, 
-                  colClasses = classes,
-                  stringsAsFactors=FALSE)
+  if(.Platform$OS.type=="unix"){
+    
+    #if file is in gzip, bz2 or zip format
+    if(grepl("^.*(.gz)[[:space:]]*$", filename)){
+      filename <- paste("gzip -dc", filename)
+    } else if (grepl("^.*(.zip)[[:space:]]*$", filename)){
+      filename <- paste("unzip -p", filename)
+    } else if (grepl("^.*(.bz2)[[:space:]]*$", filename)){  
+      filename <- paste("bzcat -dc", filename)
+    } 
+    
+    # removes the track header 
+    if(grepl("^.*(.gz|.bz2|.zip)[[:space:]]*$", filename)){
```

Comment:
it would be better if track header checking would stay as part of readGeneric not readTableFast